src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2104,14 +2110,49 @@ private String launchCommandIn(ArgumentListBuilder args, File workDir, EnvVars e
             }
             int usedTimeout = timeout == null ? TIMEOUT : timeout;
             listener.getLogger().println(" > " + command + TIMEOUT_LOG_PREFIX + usedTimeout);
-            Launcher.ProcStarter p = launcher.launch().cmds(args.toCommandArray()).
-                    envs(freshEnv).stdout(fos).stderr(err);
-            if (workDir != null) p.pwd(workDir);
-            int status = p.start().joinWithTimeout(usedTimeout, TimeUnit.MINUTES, listener);
 
-            String result = fos.toString(Charset.defaultCharset().toString());
+            int status=0;
+            String result = "";
+            String errorString = "";
+            if(ZosCheck() == true) {
+                // Another behavior on z/OS required due to the race condition
+                Launcher.ProcStarter p = launcher.launch().cmds(args.toCommandArray()).
+                        envs(freshEnv);
+                if (workDir != null) p.pwd(workDir);
+                p.readStdout().readStderr();
+                Proc prc = p.start();
+
+                status = prc.joinWithTimeout(usedTimeout, TimeUnit.MINUTES, listener);
+                BufferedReader brStdout = new BufferedReader(new InputStreamReader(prc.getStdout(),Charset.forName("IBM1047")));
```

Comment:
Please use the Java 'try with resources' syntax so that we can reduce the risk of file descriptor leaks and remove the `close` calls at the end of the block.

Reply:
Fixed.